integration_tests/cork_test.go

```diff
@@ -154,27 +147,87 @@ func (s *IntegrationTestSuite) TestCork() {
 			return true
 		}, 105*time.Second, 1*time.Second, "new vote period never seen")
 
-		s.T().Logf("checking for updated tick ranges in cellar")
+		s.T().Logf("waiting for gravity to submit call to cellar")
 		s.Require().Eventuallyf(func() bool {
-			actualTickRange, err := s.getFirstTickRange()
-			if err != nil {
-				s.T().Logf("got error %e querying ticks", err)
-				return false
+			s.T().Log("querying gravity logic call events...")
+			ethClient, err := ethclient.Dial(fmt.Sprintf("http://%s", s.ethResource.GetHostPort("8545/tcp")))
+			s.Require().NoError(err)
+
+			query := ethereum.FilterQuery{
+				FromBlock: nil,
+				ToBlock:   nil,
+				Addresses: []common.Address{
+					gravityContract,
+				},
 			}
-			if expectedTickRange.Upper != actualTickRange.Upper {
-				s.T().Logf("wrong upper %s", actualTickRange.String())
-				return false
+			logs, err := ethClient.FilterLogs(context.Background(), query)
+			s.Require().NoError(err)
+			s.T().Logf("got %v logs", len(logs))
+
+			// For non-anonymous events, the first log topic is a keccak256 hash of the
+			// event signature.
+			eventSignature := []byte("LogicCallEvent(bytes32,uint256,bytes,uint256)")
+			signatureTopic := crypto.Keccak256Hash(eventSignature).Hex()
```

Comment:
Is this something we could pull out of a generated ABI similar to what it looks like was done with the Aave contract here?

Reply:
Possibly, I'll check!

Reply:
AFAICT you you can use log filtering directly from the generated ABI but the objects you need can only be obtained if you also deploy the contract with go-ethereum. Since we are deploying with hardhat I don't think it's an option.